.github/workflows/terraform-azdev-agent-release.yml

```diff
@@ -1,69 +1,47 @@
+      - infrastructure/images/terraform-azure-devops-agent/**
+      - .github/workflows/reusable-image-scan-and-release-ghcr.yml
 
 jobs:
-  release:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - name: Get variables from tags
-        id: vars
-        run: |
-          tag=${GITHUB_REF/refs\/tags\/tf-azdev-agent-/}
-          echo "version=${tag%-demo}" >> "$GITHUB_OUTPUT"
-          echo "reponame=${GITHUB_REPOSITORY,,}" >> "${GITHUB_OUTPUT}"
-      - name: Get Git commit timestamps
-        run: echo "TIMESTAMP=$(git log -1 --pretty=%ct)" >> $GITHUB_ENV
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v3
-      - name: Login to Docker
-        uses: docker/login-action@v3
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build
-        id: docker_build
-        uses: docker/build-push-action@v6
-        with:
-          context: ./infrastructure/images/terraform-azure-devops-agent
-          push: false
-          load: true
-          platforms: linux/amd64
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
-          tags: ghcr.io/${{ steps.vars.outputs.reponame }}/terraform-azure-devops-agent:v${{ steps.vars.outputs.version }}
-        env:
-          SOURCE_DATE_EPOCH: ${{ env.TIMESTAMP }}
-      - name: Run Trivy vulnerability scanner
-        uses: aquasecurity/trivy-action@0.29.0
-        with:
-          image-ref: ghcr.io/${{ steps.vars.outputs.reponame }}/terraform-azure-devops-agent:v${{ steps.vars.outputs.version }}
-          format: 'table'
-          exit-code: '1'
-          ignore-unfixed: true
-          trivyignores: ./infrastructure/images/terraform-azure-devops-agent/.trivyignore
-          vuln-type: 'os,library'
-          severity: 'CRITICAL,HIGH'
-      - name: Push
-        id: docker_push
-        uses: docker/build-push-action@v6
-        with:
-          context: ./infrastructure/images/terraform-azure-devops-agent
-          push: true
-          platforms: linux/amd64
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
-          tags: ghcr.io/${{ steps.vars.outputs.reponame }}/terraform-azure-devops-agent:v${{ steps.vars.outputs.version }}
-        env:
-          SOURCE_DATE_EPOCH: ${{ env.TIMESTAMP }}
\ No newline at end of file
+  scan:
+    if: github.event_name == 'pull_request'
+    permissions:
+      contents: read
+      packages: read
+    uses: ./.github/workflows/reusable-image-scan-and-release-ghcr.yml
+    with:
+      release_latest: false
+      image_name: terraform-azure-devops-agent
+      tag_prefix: tf-azdev-agent-
+      workdir: ./infrastructure/images/terraform-azure-devops-agent
+      platforms: linux/amd64
+      scan_only: true
+  scan-and-release:
+    if: github.event_name != 'pull_request'
```

Comment:
nit: make the conditional more specific?
I suppose this part is going to trigger based on:
```
  push:
    tags:
      - "tf-azdev-agent-*"
```
It might make sense to make it more explicit in case we add other triggers in the future.

Reply:
This was ment as a alternativ where you differentiate for pull requests so you only give read permissions (no push from prs), but I think I like the other pattern where we give package: write even though we are only going to scan

Reply:
SO the logic here is: always scan (only give read permissions), then if it's not a PR (tag in this case) give package: wirte and push the image

Reply:
I made this the same as every other of the pipelines, then we can change it later if we see/fell a need